src/execution/riscv64/simulator-riscv64.cc

```diff
@@ -2391,7 +2394,7 @@ void Simulator::DecodeRVRFPType() {
         case 0b000: {
           if (instr_.Rs2Value() == 0b00000) {
             // RO_FMV_X_W
-            set_rd(bit_cast<int32_t>(frs1()));
+            set_rd(get_fpu_register_word(rs1_reg()));
```

Comment:
Though sign-extension is implicitly done in the original implementation. I think adding `sext_xlen` would make the sign-extension explicit (`set_rd(sext_xlen(get_fpu_register_word(rs1_reg()))`) and the code easier to understand.